src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -569,7 +591,7 @@ export default class AircraftController {
     _createAircraftWithInitializationProps(initializationProps) {
         const aircraftModel = new AircraftModel(initializationProps, this._navigationLibrary);
 
-        this.addItem(aircraftModel);
+        this._eventBus.trigger(EVENT.ADD_AIRCRAFT, aircraftModel);
```

Comment:
why is this triggering an event for a call to a local method?  are there side-effects that also need to happen now too?  those should be called out in the doc block for the `addItem` method if that is the case.

Reply:
Figured it was better to have all functions called from the event bus so it's kinda all happening from one place. I can remove and call locally here and then trigger if that's preferable?

Reply:
yeah, just all this locally.  The `EventBus` should only be used for cross-class communication.  If you just need to call a local method, do that instead of using the event bus (unless there is another class that is also listening for the event).

Reply:
> If you just need to call a local method, do that instead of using the event bus (unless there is another class that is also listening for the event).

Well that's what I'm saying though, is that there _is_ another class listening for the event. The Aircraft Controller creates the aircraft then passes the new model to the event bus with the `EVENT.ADD_AIRCRAFT` event. Listeners are the `AircraftCollection` (to add the aircraft) and the `RadarTargetCollection` (to add the radar target).

Reply:
ok.  could you add a comment to that affect?

Reply:
done